receiver/githubreceiver/logs.go

```diff
@@ -0,0 +1,315 @@
+	logEvents := r.getLogs(ctx)
+	observedTime := pcommon.NewTimestampFromTime(time.Now())
+	logs := r.processLogEvents(observedTime, logEvents)
+	if logs.LogRecordCount() > 0 {
+		if err := r.consumer.ConsumeLogs(ctx, logs); err != nil {
+			return err
+		}
+	}
+	return nil
+}
+
+func (r *gitHubLogsReceiver) getLogs(ctx context.Context) ([]gitHubEnterpriseLog) {
+	pollTime := time.Now().UTC()
+	token := r.cfg.AccessToken
+	var endpoint string
+	var curLogs []gitHubEnterpriseLog
+	var allLogs []gitHubEnterpriseLog
+	var url string
+	page := 1
+	// endpoint changes based on log type
+	if r.cfg.LogType == "user" {
+		endpoint = fmt.Sprintf("users/%s/events/public", r.cfg.Name)
+	} else if r.cfg.LogType == "organization" {
+		endpoint = fmt.Sprintf("orgs/%s/audit-log", r.cfg.Name)
+	} else {
+
+		endpoint = fmt.Sprintf("enterprises/%s/audit-log", r.cfg.Name)
+	}	
+	// Set the initial URL
+
+
+	for {
+		// Use nextURL if it's set
+		if r.nextURL != "" {
+			url = r.nextURL
+		} else {
+			url = fmt.Sprintf("https://api.github.com/%s?per_page=%d&page=%d", endpoint, gitHubMaxLimit, page) 
+		}
+
+		req, err := http.NewRequestWithContext(ctx, "GET", url, nil)
+		if err != nil {
+			r.logger.Error("error creating request: %w", zap.Error(err))
+			return allLogs
+		}
+
+		// Add headers
+		req.Header.Add("Authorization", fmt.Sprintf("Bearer %s", token))
+		req.Header.Add("Accept", "application/vnd.github.v3+json") // optional?
+		req.Header.Add("X-GitHub-Api-Version", "2022-11-28") // optional?
+		
+		resp, err := r.client.Do(req)
+		if err != nil {
+			r.logger.Error("error making request", zap.Error(err))
+			return allLogs
+		}
+		defer resp.Body.Close()
+
+		if resp.StatusCode != http.StatusOK {
+			r.logger.Error("unexpected status code", zap.Int("statusCode", resp.StatusCode))
+			return allLogs
+		}
+
+		// Read response body
+		body, err := io.ReadAll(resp.Body)
+		if err != nil {
+			r.logger.Error("error reading response body", zap.Error(err))
+			return allLogs
+		}
+		
+		// Unmarshal JSON into curLogs
+		err = json.Unmarshal(body, &curLogs)
+		if err != nil {
+			r.logger.Error("error unmarshalling JSON", zap.Error(err))
+			return allLogs
+		}
+		
+		// Append the current logs to the allLogs slice
+		allLogs = append(allLogs, curLogs...)
+		page = r.setNextLink(resp, page)
+		// Check for the 'Link' header and set the next URL if it exists
+		var curLogTime time.Time
+		if len(curLogs) != 0 {
+			curLogTime = millisecondsToTime(curLogs[len(curLogs)-1].Timestamp)
+		}
+		if r.nextURL == "" || len(curLogs) < gitHubMaxLimit || curLogTime.After(pollTime) {
+			break
+		}
+	}
+	return allLogs
+}
+
+
+
+func (r *gitHubLogsReceiver) processLogEvents(observedTime pcommon.Timestamp, logEvents []gitHubEnterpriseLog) plog.Logs {
+	logs := plog.NewLogs()
+	resourceLogs := logs.ResourceLogs().AppendEmpty()
+	resourceLogs.ScopeLogs().AppendEmpty()
+	resourceLogs.Resource().Attributes().PutStr("access_token", r.cfg.AccessToken)
```

Comment:
We probably don't want to expose the access token in the log object as it is a sensitive value.

Reply:
The log type & the name of the user/org/enterprise would be good resource attributes to add though